app/logic/questionnaire_expander.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class QuestionnaireExpander
+  class << self
+    def expand_content(content, response)
+      raise unless content.is_a? Array
+      content.reduce([]) do |total, question|
+        total.concat(question[:foreach].present? ? process_foreach(question, response) : [question])
+      end
+    end
+
+    private
+
+    def process_foreach(question, response)
+      res = []
+      case question[:foreach]
+      when :student
+        response.person.my_students.each_with_index do |person, idx|
+          res << replace_question(question.deep_dup, person, idx)
+        end
+      else
+        raise "Foreach option #{question[:foreach]} not found"
```

Comment:
Misschien "unknown" ipv "not found", of nog beter: ... `question[:foreach]` has to be `:student`.